datafusion/src/physical_plan/mod.rs

```diff
@@ -147,24 +147,59 @@ pub trait ExecutionPlan: Debug + Send + Sync {
         Distribution::UnspecifiedDistribution
     }
 
-    /// Returns `true` if the direct children of this `ExecutionPlan` should be repartitioned
-    /// to introduce greater concurrency to the plan
+    /// Returns `true` if this operator relies on its inputs being
```

Comment:
Here is the new API for `ExecutionPlan` that signal how / when repartitioning occurs